File: python_Programmers/부족한금액계산하기.py
# 내 풀이
def solution(price, money, count):
    answer = 0
    num = 0
    while count != 0:
        num += 1
        count = count - 1
        money = money - (price * num)
        answer = abs(money)
        if money > 0:
            answer = 0
    return answer

# 모범 풀이
def solution2(price, money, count):
    answer = 0
    while count > 0:
        answer += price*count
        count -=1
    return answer-money if answer-money >0 else 0

# 수학 풀이
def solution3(price, money, count):
    total = (price * (1+count) * count / 2)
    return total - money if total > money else 0

print(solution2(3, 20, ))

File: java/programmers/1단계/src/부족한금액계산하기.java
public class 부족한금액계산하기 {
    public long solution(int price, int money, int count) {
        long answer = money;
        for (int i = 1; i <= count; i++) {
            answer -= ((long) price * i);
        }
        return (answer > 0) ? 0 : -answer;
    }

    public static void main(String[] args) {
        부족한금액계산하기 sol = new 부족한금액계산하기();
        int price = 3;  // 이용금액
        int money = 20; // 잔액
        int count = 4;  // 이용횟수
        System.out.println(sol.solution(price, money, count));
    }
}
